objects/src/accounts/storage/mod.rs

```diff
@@ -19,14 +19,29 @@ pub const STORAGE_TREE_DEPTH: u8 = 8;
 // TYPE ALIASES
 // ================================================================================================
 
-/// A type that represents a single storage slot item. The tuple contains the slot index of the item
-/// and the entry of the item.
-pub type SlotItem = (u8, StorageSlot);
+/// Represents a single storage slot item.
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct SlotItem {
+    /// The index this item will occupy in the [AccountStorage] tree.
+    pub index: u8,
+
+    /// The type and value of the item.
+    pub value: StorageSlot,
+}
+
+/// Represents a single storage slot entry.
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct StorageSlot {
+    /// The type of the value
+    pub slot_type: StorageSlotType,
 
-/// A type that represents a single storage slot entry. The tuple contains the type of the slot and
-/// the value of the slot - the value can be a raw value or a commitment to the underlying data
-/// structure.
-pub type StorageSlot = (StorageSlotType, Word);
+    /// The value itself.
+    ///
+    /// The value can be a raw value or a commitment to the underlying data structure.
+    pub value: Word,
+}
```

Comment:
I wonder if we should name the fields slightly differently to avoid `value.value` situations. For example, what do you think of this?:
```Rust
pub struct SlotItem {
    /// The index this item will occupy in the [AccountStorage] tree.
    pub index: u8,

    /// The type and value of the item.
    pub slot: StorageSlot,
}

pub struct StorageSlot {
    /// The type of the value.
    pub type: StorageSlotType,

    /// The value itself.
    ///
    /// The value can be a raw value or a commitment to the underlying data structure.
    pub value: Word,
}
```

Then, the nomenclature could be `item.index`, `item.slot.type` and `item.slot.value`.

Also, I wonder if we should add some constructors to `StorageSlot` struct to simplify instantiation of storage slots. For example, something like:
```Rust
impl StorageSlot {
    pub fn new_value(value: Word) -> Self {
        Self {
            type: StorageSlotType::Value { value_arity: 0 },
            value,
        }
    }

    pub fn new_map(root: Word) -> Self {
        Self {
            type: StorageSlotType::Map { value_arity: 0 },
            value: root,
        }
    }

    pub fn new_array(root: Word) -> Self {
        Self {
            type: StorageSlotType::Array { value_arity: 0 },
            value: root,
        }
    }
}
```
In the future, we could add more constructors for the user to specify value arity too.

Reply:
changed